### PR TITLE
added testing

### PR DESCRIPTION
**Pull request by vladshender**
BookController
BookRepository
BookService
CategoryController
CategoryService

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,8 +102,34 @@
 			<version>${jjwt.version}</version>
 			<scope>runtime</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.testcontainers</groupId>
```
Comment (Elena-Bruyako) [inline pom.xml:106]: there is something with tabs, please fix it

File: src/test/java/com/example/controller/book/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                .setCategoryIds(Collections.emptySet()));
+
+        MvcResult result = mockMvc.perform(get("/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result.getResponse()
+                .getContentAsByteArray(), BookDto[].class);
+        Assertions.assertEquals(3, actual.length);
```
Comment (Elena-Bruyako) [inline src/test/java/com/example/controller/book/BookControllerTest.java:118]: use static imports

File: src/test/java/com/example/controller/book/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        MvcResult result = mockMvc.perform(put("/books/{id}", bookId)
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsByteArray(), BookDto.class);
+        Assertions.assertEquals(requestDto.getPrice(), actual.getPrice());
+    }
```
Comment (Elena-Bruyako) [inline src/test/java/com/example/controller/book/BookControllerTest.java:222]: add not valid cases

File: src/test/java/com/example/controller/category/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        MvcResult result = mockMvc.perform(put("/categories/{id}", categoryId)
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryResponseDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsByteArray(), CategoryResponseDto.class);
+        Assertions.assertEquals(expected.getName(), actual.getName());
+    }
```
Comment (Elena-Bruyako) [inline src/test/java/com/example/controller/category/CategoryControllerTest.java:200]: same here

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments, fix everywhere

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**